lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2278,10 +2278,28 @@ ParseResult FIRStmtParser::parsePrimExp(Value &result) {
       return failure();                                                        \
     }                                                                          \
     result = builder.create<CLASS>(resultTy, operands, attrs);                 \
-    return success();                                                          \
+    break;                                                                     \
   }
 #include "FIRTokenKinds.def"
   }
+  // Don't add code here, we want these two switch statements to be fused by
+  // the compiler.
```

Comment:
This is an OG Lattner comment copied from above. I personally find this a bit confusing... I assume this is saying that the body of the common cases will be merged? What might be useful for our future selves is to add a comment that this switch is here only for fixing up primops _after they have been created_.

Reply:
That is indeed what it is saying--I think it's a decent way to phrase that specific aspect of it. I am happy to add more to the comment though about fixing up primops after they've been created.